src/resmom/mom_comm.c

```diff
@@ -4719,6 +4976,108 @@ im_request(int stream, int version)
 					(void)dis_flush(efd);
 					break;
 
+				case	IM_SPAWN_MULTI:
+					/*
+					 ** Sender is MOM responding to a "spawn_multi"
+					 ** request.
+					 ** The sender MoM sent a list of task IDs
+					 ** store them on the job struct with
+					 ** the sender's node ID.
+					 ** Once we have all expected task IDs send them
+					 **
+					 ** auxiliary info (
+					 **	ident		tm_task_id;
+					 **	tvnodeid	tm_node_id;
+					 **	created		int;
+					 **	taskid 0	tm_task_id;
+					 **	...
+					 **	taskid created-1	tm_task_id;
+					 ** )
+					 */
+					DBPRT(("%s: SPAWN_MULTI %s OKAY\n",
+						__func__, jobid))
+					
+					ident = disrui(stream, &ret);
+					BAIL("OK-SPAWN_MULTI ident");
+
+					/* Get the info/pointer from the job */
+					for (tempinfo = pjob->ji_spawninfo; tempinfo != NULL; tempinfo = tempinfo->next) {
+						if (tempinfo->ji_ident == ident)
+							break;
+					}
+					if (tempinfo == NULL) {
+						sprintf(log_buffer, "Didn't find the spawninfo for the ident %d",ident);
+						log_err(-1, __func__, log_buffer);
+						break;
+					}
+					siscnt = tempinfo->ji_num_sisters;
+					index = tempinfo->ji_taskid_index;
+					temptaskid = tempinfo->ji_taskid_list;
+					if (temptaskid == NULL) {
+						goto err;
+					}
+					tempnid = tempinfo->ji_nid_list;
+					if (tempnid == NULL) {
+						goto err;
+					}
+
+					tvnodeid = disrui(stream, &ret);
+					BAIL("OK-SPAWN_MULTI tvnodeid")
+					created = disrui(stream, &ret);
+					BAIL("OK-SPAWN_MULTI created");
+					
+					/* 
+					** Store off the taskid to send all together
+					** after all tasks are created by all sisters
+					*/
+					for (i = 0; i < created; i++) {
+						taskid = disrui(stream, &ret);
+						BAIL("OK-SPAWN_MULTI taskid")
+						temptaskid[index] = taskid;
+						tempnid[index] = tvnodeid;
+						index++;
+					}
+					/* We heard back from one more sister */
+					siscnt--;
+
+					/* Don't reply until we heard back from all sisters */
+					if (siscnt != 0) {
+						tempinfo->ji_num_sisters = siscnt;
+						tempinfo->ji_taskid_index=index;
+						tempinfo->ji_taskid_list = temptaskid;
+						tempinfo->ji_nid_list = tempnid;
+						break;
+					}
+
+					/*
+					** At this point all sisters
+					** have reported back, time to reply
+					*/
+					ptask = task_check(pjob, efd, event_task);
+					if (ptask == NULL) {
+						break;
+					}
+					(void)tm_reply(efd, ptask->ti_protover,
+						TM_OKAY, event_client);
+	
+					/*
+					** First send the total number of taskids then
+					** do the write of taskid/nodeid in a loop
+					*/
+					(void)diswui(efd, index);
+					for (i = 0; i < index; i++) {
+						(void)diswui(efd, temptaskid[i]);
+						(void)diswui(efd, tempnid[i]);
+					}
+					(void)dis_flush(efd);
+					tempinfo->ji_num_sisters = 0;
+					tempinfo->ji_taskid_index = 0;
+					free(tempinfo->ji_taskid_list);
+					free(tempinfo->ji_nid_list);
+					tempinfo->ji_taskid_list = NULL;
+					tempinfo->ji_nid_list = NULL;
```

Comment:
Should you free tempinfo here?

Reply:
No, tempinfo is freed when the job structure is freed as part of job_free()